### PR TITLE
Remove problem+json Content-Type

### PR DESCRIPTION
**Pull request by jbagga**
Addresses #7344

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorOptionsSetup.cs [hunk 1]
```diff
@@ -36,8 +36,8 @@ IActionResult GetInvalidModelStateResponse(ActionContext context)
                     new BadRequestObjectResult(modelState) :
                     new BadRequestObjectResult(errorDetails);
 
-                result.ContentTypes.Add("application/problem+json");
-                result.ContentTypes.Add("application/problem+json");
+                result.ContentTypes.Add("application/json");
+                result.ContentTypes.Add("application/json");
```
Comment (khellang) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorOptionsSetup.cs]: This duplicate should be removed, no? I think I raised this with @rynowak earlier, but can't find the comment...

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorOptionsSetup.cs [hunk 2]
```diff
@@ -29,15 +29,14 @@ public void Configure(ApiBehaviorOptions options)
...
-                    context.ActionDescriptor, 
+                    context.ActionDescriptor,
                     context.ModelState);
 
                 var result = (errorDetails is ModelStateDictionary modelState) ?
                     new BadRequestObjectResult(modelState) :
                     new BadRequestObjectResult(errorDetails);
 
                 result.ContentTypes.Add("application/json");
-                result.ContentTypes.Add("application/json");
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorOptionsSetup.cs]: Think this was meant to be `application/xml`

Comment (jbagga) [reply src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorOptionsSetup.cs]: Okay. I didn't expect that

**Comment by jbagga** [comment]
@pranavkm Updated

**Comment by pranavkm** [comment]
:shipit: once test pass

**Pull request merged**